docs-src/concepts.md

```diff
@@ -112,42 +114,123 @@ TODO
+defined by the infrastructure ops. This resource represents a class of Gateways
+that can be instantiated. 
+
+[gatewayclass-src]: https://github.com/kubernetes-sigs/service-apis/blob/master/api/v1alpha1/gatewayclass_types.go
+
+> Note: this serves the same function as the `networking.IngressClass` resource.
 
 ```yaml
-apiVersion: networking.x-k8s.io/v1alpha1
 kind: GatewayClass
 metadata:
-  name: from-internet
-controller:                          # links to a custom resource
-  apiGroup: networking.acme.io       # that implements this class. 
-  kind: cloud-lb
-  name: from-internet
+  name: cluster-gateway
+spec:
+  controller: "acme.io/gateway-controller"
 ```
 
-A sample controller file:
+We expect that one or more `GatewayClasses` will be created by the infrastructure
+provider for the user. It allows decoupling of which mechanism (e.g. controller)
+implements the `Gateways` from the user. For instance, an infrastructure
+provider may create two `GatewayClasses` named `internet`, `private` to reflect
+`Gateways` that define Internet-facing vs private, internal applications.
 
 ```yaml
-apiVersion: networking.acme.io/v1alpha1
-kind: CloudLB
+kind: GatewayClass
+metadata:
+  name: internet
+  ...
+---
+kind: GatewayClass
 metadata:
-  name: from-internet
-networkTier: auto
-ipAllocation: auto
+  name: private
+  ...
 ```
 
-We expect that one or more GatewayClasses will be created by the infrastructure
-provider for the user as part of their cloud infrastructure. If the cluster
-operator wishes to customize their `GatewayClasses`, they are free to do so,
-interacting with the custom resource parameters. Example parameters
-`GatewayClass.controller` resource will include cluster-level defaults and
-constraints.
+The user of the classes will not need to know *how* `internet` and `private` are
+implemented, whether they are cluster-based or cloud-based LBs.
 
 > The community may converge around a common set of GatewayClass names with
 > well-known meanings (e.g. `internet`, `private`) to ease interoperability
 > between platforms.
 
+#### GatewayClass parameters
+
+Providers of the `Gateway API` may need to pass parameters to their controller
+as part of the class definition. This done using the
+`GatewayClass.spec.parameters` field:
+
+```yaml
+# GatewayClass for Gateways that define Internet-facing applications.
+kind: GatewayClass
+metadata:
+  name: internet
+spec:
+  controller: "acme.io/gateway-controller"
+  parameters:
+    apiVersion: core/v1
+    kind: ConfigMap
+    namespace: acme-system
+    name: internet-gateway
+---
+kind: ConfigMap
+metadata:
+  name: internet-gateway
+  namespace: acme-system
+data:
+  ip-address-pool: internet-vips
+  ...
+```
+
+The type of object referenced by `GatewayClass.spec.parameters` will depend on
+the provider itself. A `core.ConfigMap` is used in the example above, but some
+controllers may opt to use a `CustomResource` for better schema validation.
+
+#### GatewayClass status
```

Comment:
To be clear... the only prescriptive condition for implementations is for parameter validation? When validity is observed, what else is implied about the gateway controller implementation? Or, what does it mean when these conditions are observed and what can I infer from or do about them?

For example, do validated parameters imply that the controller is now functioning and should be reconciling Gateways?

Reply:
Yes -- given that we have configuration parameters in the GatewayClass, there needs to be some way to get status back to the user, especially if this is the admin that is creating it.

There is an alternative: put the status in the Gateway when that is created.

It makes it harder for admins to set up GatewayClass once, know that their settings work; they have to  create a test Gateway etc in that case.

Reply:
I think the answer is yes, at this point. Remember as well that the point of Conditions being an array is that more can be added if required.

Reply:
yeah -- conditions is the standard way to do an open ended enum (see API conventions)